packages/@react-spectrum/color/chromatic/ColorArea.chromatic.tsx

```diff
@@ -0,0 +1,196 @@
+);
+
+let difference = (a, b): Set<ColorChannel> => new Set([...a].filter(x => !b.has(x)));
+
+function ColorAreaExample(props: SpectrumColorAreaProps) {
+  let {xChannel, yChannel, isDisabled} = props;
+  let defaultValue = typeof props.defaultValue === 'string' ? parseColor(props.defaultValue) : props.defaultValue;
+  let [color, setColor] = useState(defaultValue || parseColor('#ff00ff'));
+  let xyChannels = new Set([xChannel, yChannel]);
+  let colorSpace = color.getColorSpace();
+  let zChannel: ColorChannel = difference(color.getColorChannels(), xyChannels).keys().next().value;
+  let isHue = zChannel === 'hue';
+
+  function onChange(e) {
+    try {
+      e = normalizeColor(e);
+      // eslint-disable-next-line no-empty
+    } catch (error) {
+      e = undefined;
+      return;
+    }
+    const newColor = (e || color).toFormat(colorSpace);
+    if (props.onChange) {
+      props.onChange(newColor);
+    }
+    setColor(newColor);
+  }
+
+  return (
+    <div role="group" aria-label={`${colorSpace.toUpperCase()} Color Picker`}>
+      <Flex gap="size-500" alignItems="start">
+        <Flex direction="column" gap={isHue ? 0 : 'size-50'} alignItems="center">
+          <ColorArea
+            size={isHue ? 'size-1200' : null}
+            {...props}
+            value={color}
+            onChange={onChange}
+            onChangeEnd={props.onChangeEnd} />
+          {isHue ? (
+            <ColorWheel
+              value={color}
+              onChange={onChange}
+              onChangeEnd={props.onChangeEnd}
+              isDisabled={isDisabled}
+              size={'size-2400'}
+              UNSAFE_style={{
+                marginTop: 'calc( -.75 * var(--spectrum-global-dimension-size-2400))'
+              }} />
+          ) : (
+            <ColorSlider
+              value={color}
+              onChange={onChange}
+              onChangeEnd={props.onChangeEnd}
+              channel={zChannel}
+              isDisabled={isDisabled} />
+          )}
+        </Flex>
+        <Flex direction="column" alignItems="center" gap="size-100" minWidth="size-1200">
+          <div
+            role="img"
+            aria-label={`color swatch: ${color.toString('rgb')}`}
+            title={`${color.toString('hex')}`}
+            style={{width: '96px', height: '96px', background: color.toString('css')}} />
+          <ColorField
+            label="HEX Color"
+            value={color}
+            onChange={onChange}
+            onKeyDown={event =>
+              event.key === 'Enter' &&
+              onChange((event.target as HTMLInputElement).value)
+            }
+            isDisabled={isDisabled}
+            width="size-1200" />
+        </Flex>
+      </Flex>
+    </div>
+  );
+}
+
+export let XBlueYGreen = Template.bind({});
+XBlueYGreen.storyName = 'RGB xChannel="blue", yChannel="green"';
+XBlueYGreen.args = {xChannel: 'blue', yChannel: 'green', onChange: action('onChange'), onChangeEnd: action('onChangeEnd')};
```

Comment:
why are you calling `action()` in a chromatic story?

Reply:
it generates a function